src/astro/databases/base.py

```diff
@@ -461,10 +475,50 @@ def load_file_to_table_natively(
         source_file: File,
         target_table: Table,
         if_exists: LoadExistStrategy = "replace",
+        native_support_kwargs: Optional[Dict] = None,
         **kwargs,
     ):
         """
         Checks if optimised path for transfer between File location to database exists
-        and if it does, it transfers it and returns true else false.
+        and if it does, it transfers it and returns true else false
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        :param if_exists: Overwrite file if exists. Default False
+        :param native_support_kwargs: kwargs to be used by method involved in native support flow
         """
         raise NotImplementedError
+
+    def create_empty_table(
+        self, source_file: File, target_table: Table, chunksize: int = 1000
+    ):
+        """
+        Infer schema from source file and create and empty table in database
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        :param chunksize: No. of rows to use to infer schema
+        """
+        df = source_file.export_to_dataframe(chunksize=chunksize)
+        # When we pass chunksize we get reference to data and not the data.
+        # We need to use read() to get the 1st chunk.
+        df = df.read()
+
+        self.load_pandas_dataframe_to_table(
+            source_dataframe=df,
+            target_table=target_table,
+            if_exists="replace",
+            chunk_size=chunksize,
+        )
+        self.truncate_table(target_table)
```

Comment:
It may be cleaner to use something like the following to create an empty table:
```
from pandas.io.sql import get_schema

schema_statement = get_schema(df, self.get_table_qualified_name(target_table), con=self.sqlalchemy_engine)
self.run_sql(schema_statement)
```
As opposed to loading stuff and having to delete them afterwards.

Reply:
We need a unit test for this too

Reply:
Added unit test.